tasks/rh-sign-cosign-image/rh-sign-cosign-image.yaml

```diff
@@ -0,0 +1,104 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: rh-sign-cosign-image
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Task to sign container images in a Snapshot spec using the cosign.
+  params:
+    - name: snapshotPath
+      description: Path to the JSON string of the mapped Snapshot spec in the data workspace
+      type: string
+    - name: dataPath
+      description: Path to the JSON string of the merged data to use in the data workspace
+      type: string
+    - name: secretName
+      description: Name of secret containing needed credentials 
+      type: string
+  workspaces:
+    - name: data
+      description: workspace to read and save files
+  steps:
+    - name: sign-image
+      image: quay.io/konflux-ci/release-service-utils:38c3bfd7479c86b832cba5b61f9bbde40c469393
+      env:
+        - name: AWS_REGION
+          valueFrom:
+            secretKeyRef:
+              name: $(params.secretName)
+              key: AWS_REGION
+        - name: AWS_ACCESS_KEY_ID
+          valueFrom:
+            secretKeyRef:
+              name: $(params.secretName)
+              key: AWS_ACCESS_KEY_ID
+        - name: AWS_SECRET_ACCESS_KEY
+          valueFrom:
+            secretKeyRef:
+              name: $(params.secretName)
+              key: AWS_SECRET_ACCESS_KEY
+      script: |
+        #!/usr/bin/env bash
+        set -eu
+        function fetch_cosign_key() {
+          kubectl get configmap --output json | jq -r ".data.SIG_KEY" | tr -d \"
```

Comment:
I don't think this will work. The name of the configmap to fetch is missing - I guess it should be the one from the data json. But overall I'm still not sure about this - I think this should be a secret. And because of this (and the aws stuff) I think we'll also need a "prepare-cosign-sign-parameters" task like we have for pyxis stuff and now also for the cdn tasks that Scott W. was working on - @johnbieren  you know what I'm talking about, right? WDYT?

(Also, the `tr` seems redundant here.)

Reply:
Changed in https://github.com/konflux-ci/release-service-catalog/pull/467. Key is got from secret now.

Reply:
Thanks.